apps/seqan_flexbar/general_processing.h

```diff
@@ -483,474 +483,240 @@ void processN(TSeqs& seqs, TIds& ids, TSeqs& seqsRev, TIds& idsRev, TMulti& mult
-                erase(seqsRev[i], length(seqsRev[i]) - tail, length(seqsRev[i]));
-                if ((length(seqs[i]) >= min) && (length(seqsRev[i]) >= min)) 
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (head > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if ((length(seqs[i]) > head) && (length(seqsRev[i]) > head))
-            {
-                erase(seqs[i], 0, head);
-                erase(seqsRev[i], 0, head);
-                if ((length(seqs[i]) >= min) && (length(seqsRev[i]) >= min))
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if ((length(seqs[i]) > tail) && (length(seqsRev[i]) > tail))
-            {
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                erase(seqsRev[i], length(seqsRev[i]) - tail, length(seqsRev[i]));
-                if ((length(seqs[i]) >= min) && (length(seqsRev[i]) >= min))
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if ((length(seqs[i]) >= min) && (length(seqsRev[i]) >= min))
-            {
-                rem[i] = false;
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    unsigned ex = 0;
-    for (int j = length(rem) - 1; j >= 0; --j)
-    {
-        if (rem[j])
-        {
-            swap(seqs[j], seqs[limit - ex - 1]);
-            swap(seqsRev[j], seqsRev[limit - ex - 1]);
-            swap(ids[j], ids[limit - ex - 1]);
-            swap(idsRev[j], idsRev[limit - ex - 1]);
-            ++ex;
-        }
-    }
-    if (ex != 0)
-    {
-        resize(seqs, limit - ex);
-        resize(seqsRev, limit - ex);
-        resize(ids, limit - ex);
-        resize(idsRev, limit - ex);
-        stats.removedSeqsShort += (2 * ex);
-    }
+	StringSet<bool> rem;
+	_preTrim(seqs, ids, head, nexus, tail, min, rem);
+	std::vector<TSeqs*> seqsVector;
```

Comment:
Because then I can use the same _preTrimRemove function, for single and double strands.

For double strands I do:
	seqsVector.emplace_back(&seqs);
	seqsVector.emplace_back(&seqsRev);
	idsVector.emplace_back(&ids);
	idsVector.emplace_back(&idsRev);
	_preTrimRemove(seqsVector, idsVector, rem1, demultiplexParams, stats);

Actually it would be better to make a vector of THasIteratorObjects, then I could do:
	objectsVector.emplace_back(&seqs);
	objectsVector.emplace_back(&seqsRev);
	objectsVector.emplace_back(&ids);
	objectsVector.emplace_back(&idsRev);
        if (demultiplex)
              objectsVector.emplace_back(&demultiplexParams);
	_preTrimRemove(seqsVector, idsVector, rem1, demultiplexParams, stats);

and the _preTrimRemove would also be much simpler. Is there a common Iterator-Class for seqan::StringSet's ?